code/__HELPERS/unsorted.dm

```diff
@@ -527,7 +527,8 @@ Turf and target are separate in case you want to teleport some distance from a t
 		//Byond does not allow things to be in multiple contents, or double parent-child hierarchies, so only += is needed
 		//This is also why we don't need to check against assembled as we go along
 		processing_list += A.contents
-		assembled += A
+		if(!T || istype(A,T))
```

Comment:
This is a fairly performance-sensitive proc so I recommend making a different version like GetAllContentsOfType or something, alternatively put your if before the while and have a different loop based on whether you're looking for specific types only or not.

Reply:
Ok